src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs

```diff
@@ -48,8 +48,15 @@ private void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
                 // out-vars are not supported prior to C# 7.0.
                 return;
             }
-
-            var optionSet = context.Options.GetOptionSet();
+            var options = context.Options;
+            var syntaxTree = context.Node.SyntaxTree;
+            var cancellationToken = context.CancellationToken;
+            var optionSet = options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
```

Comment:
lots of duplication across all our analyzers.  Can you have a single extnsion method that does this?